### DISCUSSION TITLE
How to use Android API?

### DISCUSSION THREAD

**Discussion by userjlegwork** [beeware/beeware#199] [Q&A] [answered]
I started a project for school using beeware, that project is intended to compile to windows and android from the same codebase (this is accomplished very well by beeware) but the android part needs to access the android API to get some permissions (using the 'android' module), but when I do the `import android` it says "module android not found", I did it at the end checking if the app is running on Android and then doing the import, because I noticed that that module can be imported inside the boilerplate generated on the build/android directory but that code is generated by beeware based on the project code. So, short version, how to develop the app for windows/android from the same codebase? 🙂 Thanks in advance.

**Comment by freakboy3742** [comment, marked as answer]
If you add any code that is platform-specific, you need to "gate" that logic with conditional platform checks. 

For example, if you're using Toga, you can write:
```
if toga.current_platform == "android":
    import android
    # do android things
elif toga.current_platform == "windows":
    import some_windows_library
    # do windows things
```

If you have a lot of platform-specific code, you may find it helpful to put all your platform-specific logic into a submodule (e.g., `myapp.windows`, `myapp.android`), and use `importlib.import_module()` to import the specific module you need based on the value of `toga.current_platform`. If all your platform submodules have the same functions defined (but with platform-specific implementations), you can effectively use the module as an interface to swap in different behaviours. This is effectively what Toga itself does - each platform backend is expected to meet a basic functional API, and the core Toga interface can call on those modules without caring which platform it is on.

**Reply by userjlegwork** [reply]
Thanks a lot, this is exactly what I was looking for 😄🤙.
